precompile/assetsbridge/bridge_out.go

```diff
@@ -0,0 +1,615 @@
+package assetsbridge
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/btcsuite/btcutil"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/tracing"
+	"github.com/holiman/uint256"
+	"github.com/mezo-org/mezod/precompile"
+	bridgetypes "github.com/mezo-org/mezod/x/bridge/types"
+	evmtypes "github.com/mezo-org/mezod/x/evm/types"
+)
+
+// BridgeOutMethodName is the name of the bridgeOut method.
+// It matches the name of the method in the contract ABI.
+//
+//nolint:gosec
+const BridgeOutMethodName = "bridgeOut"
+
+// SendMsgURL defines the authorization type for MsgSend
+var SendMsgURL = sdk.MsgTypeURL(&banktypes.MsgSend{})
+
+// BridgeOutMethod is the implementation of the getCurrentSequenceTip method.
+type BridgeOutMethod struct {
+	bridgeKeeper BridgeKeeper
+	bankKeeper   BankKeeper
+	evmKeeper    EvmKeeper
+	authzKeeper  AuthzKeeper
+}
+
+func newBridgeOutMethod(
+	bridgeKeeper BridgeKeeper,
+	bankKeeper BankKeeper,
+	evmKeeper EvmKeeper,
+	authzKeeper AuthzKeeper,
+) *BridgeOutMethod {
+	return &BridgeOutMethod{
+		bridgeKeeper: bridgeKeeper,
+		bankKeeper:   bankKeeper,
+		evmKeeper:    evmKeeper,
+		authzKeeper:  authzKeeper,
+	}
+}
+
+func (m *BridgeOutMethod) MethodName() string {
+	return BridgeOutMethodName
+}
+
+func (m *BridgeOutMethod) MethodType() precompile.MethodType {
+	return precompile.Write
+}
+
+func (m *BridgeOutMethod) RequiredGas(_ []byte) (uint64, bool) {
+	// Fallback to the default gas calculation.
+	return 0, false
+}
+
+func (m *BridgeOutMethod) Payable() bool {
+	return false
+}
+
+func (m *BridgeOutMethod) Run(
+	context *precompile.RunContext,
+	rawInputs precompile.MethodInputs,
+) (precompile.MethodOutputs, error) {
+	// extract inputs
+	inputs, err := m.extractInputs(rawInputs)
+	if err != nil {
+		return precompile.MethodOutputs{false}, err
+	}
+
+	// run validation
+	if err := m.validate(context, inputs); err != nil {
+		return precompile.MethodOutputs{false}, err
+	}
+
+	return m.execute(context, inputs)
+}
+
+// execute will execute the burn of the token then send
+// AssetUnlocked events to the bridgeKeeper.
+func (m *BridgeOutMethod) execute(
+	context *precompile.RunContext,
+	inputs *bridgeOutInputs,
+) (precompile.MethodOutputs, error) {
+	var (
+		err           error
+		assetUnlocked *bridgetypes.AssetUnlockedEvent
+	)
+
+	switch inputs.Chain {
+	case TargetChainEthereum:
+		assetUnlocked, err = m.executeEthereum(context, inputs)
+	case TargetChainBitcoin:
+		assetUnlocked, err = m.executeBitcoin(context, inputs)
+	}
```

Comment:
BTC can be bridged out to [both Ethereum (as TBTC) and Bitcoin](https://linear.app/thesis-co/document/technical-architecture-9fbdf2ac111c#heading-validation-logic-16c41930). This condition does not account for this fact.